cmd/helpers.go

```diff
@@ -89,17 +132,86 @@ func BuildApplication(binaryName string, forceRebuild bool, buildMode string, pa
 		println(compileMessage)
 	}
 
+	// embed resources
+	targetFiles, err := EmbedAssets()
+	if err != nil {
+		return err
+	}
+
+	// cleanup temporary embedded assets
+	defer func() {
+		for _, filename := range targetFiles {
+			if err := os.Remove(filename); err != nil {
+				fmt.Println(err)
+			}
+		}
+	}()
+
+	// Setup ld flags
+	ldflags := "'-w -s "
+	if buildMode == BuildModeDebug {
+		ldflags = ""
+	}
+
+	// Add windows flags
+	if projectOptions.Platform == "windows" && buildMode == BuildModeProd {
+		ldflags += "-H windowsgui "
+	}
+
+	ldflags += "-X github.com/wailsapp/wails.BuildMode=" + buildMode
+
+	// If we wish to generate typescript
+	if projectOptions.typescriptDefsFilename != "" {
+		cwd, err := os.Getwd()
+		if err != nil {
+			return err
+		}
+		filename := filepath.Join(cwd, projectOptions.FrontEnd.Dir, projectOptions.typescriptDefsFilename)
+		ldflags += " -X github.com/wailsapp/wails/lib/binding.typescriptDefinitionFilename=" + filename
+	}
+	ldflags += "'"
+
 	buildCommand := slicer.String()
-	buildCommand.Add("mewn")
+	if projectOptions.CrossCompile {
+		user, err := user.Current()
+		if err != nil {
+			return err
+		}
+		for _, arg := range []string{
+			"docker",
+			"run",
+			"--rm",
+			"-v", fmt.Sprintf("%s/build:/build", fs.Cwd()),
+			"-v", fmt.Sprintf("%s:/source", fs.Cwd()),
+			"-e", fmt.Sprintf("LOCAL_USER_ID=%v", user.Uid),
+			"-e", fmt.Sprintf("FLAG_LDFLAGS=%s", ldflags),
+			"-e", "FLAG_V=false",
+			"-e", "FLAG_X=false",
+			"-e", "FLAG_RACE=false",
+			"-e", "FLAG_BUILDMODE=default",
+			"-e", "FLAG_TRIMPATH=false",
+			"-e", fmt.Sprintf("TARGETS=%s", projectOptions.Platform+"/"+projectOptions.Architecture),
+			"-e", "GOPROXY=",
+			"-e", "GO111MODULE=on",
```

Comment:
Am wondering if we need GOPRIVATE... Haven't really thought about it before.

Reply:
Probably do.  I have a good way of dealing with that.  Probably should make an `issue` to cover it.

Reply:
> Linux->Mac build:
> 
>     * Looks like the bundled app is being written to the project root. Are we expecting it to be in the build directory?
> 

I meant to change where that was being created and update the path to move it in to the `build` directory.